applications/tari_app_utilities/src/commands.rs

```diff
@@ -0,0 +1,373 @@
+// Copyright 2020. The Tari Project
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use crate::command_parser::{ParsedArgument, ParsedCommand};
+use chrono::{DateTime, Utc};
+
+use log::*;
+use std::{str::FromStr, time::Duration};
+use strum_macros::{Display, EnumIter, EnumString};
+use tari_common::GlobalConfig;
+use tari_core::transactions::tari_amount::uT;
+use tari_wallet::{
+    transaction_service::{
+        error::TransactionServiceError,
+        handle::{TransactionEvent, TransactionServiceHandle},
+    },
+    WalletThreadSafe,
+};
+use thiserror::Error;
+use tokio::{
+    runtime::Handle,
+    stream::StreamExt,
+    task::JoinError,
+    time::{delay_for, timeout},
+};
+
+/// Enum representing commands used by the wallet
+#[derive(Clone, PartialEq, Debug, Display, EnumIter, EnumString)]
+#[strum(serialize_all = "kebab_case")]
+pub enum WalletCommand {
+    GetBalance,
+    SendTari,
+    MakeItRain,
+}
+
+#[derive(Debug, Error)]
+pub enum CommandError {
```

Comment:
Nit: but we tend to put these in their own error.rs file so they are easy to find. This will be easier when moving the logic in this file into a module in the `tari_console_wallet` app I think.

Reply:
Cool - will do